R/commonly_used_db_data.R

```diff
@@ -0,0 +1,277 @@
+                 "fvps_source", "fvps_adjusted", "coverage_adjusted")]
+  print("Transformed coverage data.")
+  
+  country_id <- DBI::dbReadTable(con, "country")
+  cov2$country_nid <- country_id$nid[match(cov2$country, country_id$id)]
+  return(cov2[order(cov2$delivery_id), ])
+  ### this function gives differnet fvps_adjusted comparing to fix_coverage_fvps()
+  ### I am sure this function is more accurate, as fix_coverage_fvps() is a bit buggy here
+  ### https://gitbub.com/vimc/jenner/blob/master/R/impact_method2.R#L419
+  ### see measles for example (bdi, 2015)
+}
+
+
+
+##' Extract demographic data
+##' 
+##' @title Extract demographic data
+##' 
+##' @param con Datebase connection
+##' @param touchstone_pop Demography touchstone
+##' @param demographic_statistic Demographic statistic to extract
+##' @param gender Gender codes - "Male", "Female", "Both"
+##' @param country_ All countries if NULL. Or specify a vector of countries
+##' @param year_ All years if NULL. Or specify a vector of years
+##' @param age_ All age groups if NULL. Or specify a vector of age groups
+##' @export
+get_population <- function(con, touchstone_pop = "201710gavi-5", demographic_statistic = "int_pop", gender = "Both",
+                           country_ = NULL, year_ = NULL, age_ = NULL) {
+  if(!grepl("-", touchstone_pop)){
+    touchstone_pop <- get_touchstone(con, touchstone_pop)
+    message("touchstone version is not specified. Lateset version is used.")
+  }
+  ## this table get population data as you wish
+  sql <- read_sql("inst/sql/population.sql")
+  constrains_ <- sql_constrains(country_, year_, age_)
+  sql <- sprintf(sql, sql_in(gender), constrains_)
+  return(DBI::dbGetQuery(con, sql, list(touchstone_pop, demographic_statistic)))
+  
+}
+
+## supporting function for get_population()
+## flexibally contrain country, year, and age for get_population()
+## currently burden_estimate_table parameter is set to be FALSE
+## when it is set to be true
+## it is supposed to constrain country, year, and age for reading burden estimates
+## cannot remember where it's been used
+sql_constrains <- function(country_ = NULL, year_ = NULL, age_ = NULL, burden_estimate_table = FALSE) {
+  country_id <- ifelse(burden_estimate_table, "country.id", "country")
+  age_id <- ifelse(burden_estimate_table, "age", "age_from")
+  
+  country_ <- ifelse(is.null(country_),
+                     "\t",
+                     sprintf("AND %s IN %s",country_id, sql_in(country_)))
+  
+  year_ <- ifelse(is.null(year_),
+                  "\t",
+                  sprintf("AND year IN %s", sql_in(year_, text_item = FALSE)))
+  
+  age_ <- ifelse(is.null(age_),
+                 "\t",
+                 sprintf("AND %s IN %s", age_id, sql_in(age_, text_item = FALSE)))
+  
+  return(paste(country_, year_, age_, collapse = "\n"))
+}
+
+### all cause mortality by birth cohort - for the papert
+##' Extract demographic data - all couse mortality, live birth and under5 mortality rate
+##' 
```

Comment:
I think this would be clearer if docs expanded on what it does and we include better explanation on the params, also need an `@return` section. Would be good to include some examples too.